cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java

```diff
@@ -223,37 +225,58 @@ public Void call() throws Exception {
     metadataStore.removeMetadata(datasetInstanceId);
   }
 
+  public boolean exists(DatasetId datasetInstanceId) throws Exception {
+    return performDatasetAdmin(datasetInstanceId, new Operation<Boolean>() {
```

Comment:
Using method reference would be cleaner:

```
return performDatasetAdmin(datasetInstanceId, DatasetAdmin::exists);
```

Reply:
Oh right we only support Java8 now. Good point.